components/feature/qr/src/main/java/mozilla/components/feature/qr/QrFragment.kt

```diff
@@ -495,13 +515,38 @@ class QrFragment : Fragment() {
                         logger.error("Failed to configure CameraCaptureSession")
                     }
                 }
-                // Deprecation of createCaptureSession() will be handled in https://github.com/mozilla-mobile/android-components/issues/8510
-                @Suppress("DEPRECATION")
-                it.createCaptureSession(Arrays.asList(mImageSurface, surface), stateCallback, null)
+                createCaptureSessionCompat(it, mImageSurface as Surface, surface, stateCallback)
             }
         }
     }
 
+    @VisibleForTesting
+    internal fun createCaptureSessionCompat(
+        camera: CameraDevice,
+        imageSurface: Surface,
+        surface: Surface,
+        stateCallback: CameraCaptureSession.StateCallback
+    ) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+            if (shouldStartExecutorService()) {
+                startExecutorService()
+            }
+            val sessionConfig = SessionConfiguration(
+                SessionConfiguration.SESSION_REGULAR,
+                listOf(OutputConfiguration(imageSurface), OutputConfiguration(surface)),
+                backgroundExecutor as Executor,
+                stateCallback
+            )
+            camera.createCaptureSession(sessionConfig)
+        } else {
+            @Suppress("DEPRECATION")
+            camera.createCaptureSession(listOf(imageSurface, surface), stateCallback, null)
+        }
+    }
```

Comment:
@levimatheri I create this separate function to test this section by separated without having to provide all the previous dependencies.

Reply:
Aha, that makes sense. Thanks for the link too. Do you have a guide on how to rebase/squash commits properly for PRs?

Reply:
I think [this](https://thoughtbot.com/blog/git-interactive-rebase-squash-amend-rewriting-history) could helpful  :)